tools/upload_buildomat_junit_to_trunk.sh

```diff
@@ -0,0 +1,137 @@
+#!/usr/bin/env bash
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+# We use Trunk (https://trunk.io) to track flaky tests in our suite. Trunk uses a custom CLI to
+# upload JUnit XML files, but unfortunately we cannot run it in Buildomat, as it requires a secret.
+#
+# To work around that, we use this script to grab the JUnit XML file from a finished Buildomat job
+# and upload it. The script requires the GitHub Check Run ID of the Buildomat job, and takes care of
+# retrieving the rest of the metadata from the GitHub API.
+#
+# While the script is meant to be executed in a GitHub Actions workflow triggering after the
+# Buildomat job (thanks to the check_run event), it is possible to run the script locally (with a
+# valid GITHUB_TOKEN, TRUNK_ORG_SLUG and TRUNK_TOKEN).
+
+set -euo pipefail
+IFS=$'\n\t'
+
+if [[ $# -ne 1 ]]; then
+    echo "usage: $0 <check_run_id>" >&2
+    exit 1
+fi
+check_run_id="$1"
+
+tmp="$(mktemp -d)"
+trap "rm -rf ${tmp}" EXIT
+cd "${tmp}"
+
+mkdir api/
+
+gh api "repos/${GITHUB_REPOSITORY}/check-runs/${check_run_id}" > api/check_run
+check_suite_id="$(jq -r .check_suite.id api/check_run)"
+commit="$(jq -r .head_sha api/check_run)"
+github_app="$(jq -r .app.slug api/check_run)"
+job_url="$(jq -r .details_url api/check_run)"
+pr_url="$(jq -r .pull_requests[0].url api/check_run)"
```

Comment:
Can there be multiple PRs attached to a check run?

Reply:
Yes.

PRs cannot be attached to check runs. Instead, GitHub tries to look for PRs that contain the branch being tested, and includes them in the API response. From their docs:

> Pull requests that are open with a `head_sha` or `head_branch` that matches the check. The returned pull requests do not necessarily indicate pull requests that triggered the check

In practice, if there are multiple PRs returned by the API this will show in the Trunk UI just the information of the first one. I don't think it's the end of the world though, I don't expect it to happen much.